File: order-react/src/type/order.ts
export interface Order {
    id: number;
    orderNumber: string;
    totalAmount: number;
    orderDate: string;
    recipient: string;
    deliveryAddress: string;
    paymentType: 'карта' | 'наличные';
    deliveryType: 'самовывоз' | 'доставка до двери';
    orderDetails: OrderDetail[];
}

export interface OrderDetail {
    id: number;
    productCode: number;
    productName: string;
    quantity: number;
    unitPrice: number;
    orderId: number;
}
export interface FetchOrderByTotalArgs {
    startDate: string;
    minTotalAmount: number;
}

File: orderservice/domain/src/main/java/com/itqgroup/entity/Order.java
package com.itqgroup.entity;

import lombok.AllArgsConstructor;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.Getter;
import lombok.ToString;

import java.util.List;
import java.time.LocalDate;

/**
 * Класс сущность для заказа.
 */
@Getter
@Setter
@ToString
@NoArgsConstructor
@AllArgsConstructor
public class Order {
    private Long id;
    private String orderNumber;
    private Double totalAmount;
    private LocalDate orderDate;
    private String recipient;
    private String deliveryAddress;
    private String paymentType;
    private String deliveryType;
    private List<OrderDetail> orderDetails;
}
